### PR TITLE
docs: update contribution guidelines

### LINKED ISSUE #23: Fix contribution guidelines
## Description

The [current contribution guidelines](https://github.com/digipolisantwerp/acpaas-ui_angular/blob/master/CONTRIBUTING.md) are copied from vue.js, and contain various things which are not correct:

- refers to stackoverflow to post questions, which nobody is keeping an eye on
- proposes that PR's are sent to the dev branch, but no dev branch exists
- is missing guidelines around examples and documentation
- refers to "npm run commit" which doesn't exist
- does not explain the project structure
- is missing an explanation of how releases happen (how do PR's end up in a release)

## Proposed changes

- Where to ask questions:
    - Slack for quick questions
    - Github issues for bug reports or feature requests
    - ~~Google groups for longer discussions (a new group is set up, we offload these discussions here so they are permanently documented and do not pollute the issue space)~~ (github issues for everything, see comments)
- PR steps / checklist
    - Fork + topic branch
    - Proper branch (see release model)
    - Commit message format (remove mention of npm run commit)
    - Features are accompanied by documentation updates and unit tests (or explain in the PR why docs / tests are not necessary)
- Release model
    - PR's to master, not to dev, because merging regularly from dev to master is going to be supremely annoying, and not really any more flexible or stable. Stick close to github flow model.
    - Tag to release straight from master. Hotfixes can be done by branching off the tag, cherry-picking from master and tagging from the branch (+ publishing the tag to npm). This is as compared to branching for release every time, which is a lot more work without a real practical benefit.
    - When doing a major release, create a maintenance branch for it (e.g. v1-dev), from which maintenance releases can be done.
    - Release frequency: ad-hoc for high-prio changes, otherwise monthly. (+ describe strategy for projects that need upstream changes so they don't have to wait a month)

### PR DESCRIPTION
**Pull request by jsebrech**
## PR Checklist

- [x] The commit message follows our Contributing guidelines

## PR Type

Changes to the contribution guide and assorted markdown files, as described in #23. Added a policy issue template. Added issue handling guidelines. Made style changes to markdown files to appease the markdownlint gods.

Recommended review approach: start at CONTRIBUTING.md and work your way through it, following the links to other markdown files where they occur.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

## Resolved issues

Closes #23

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -2,78 +2,97 @@
...
+  - [Issue Reporting Guidelines](#issue-reporting-guidelines)
+  - [Pull Request Guidelines](#pull-request-guidelines)
+  - [Development Setup](#development-setup)
+  - [Project Structure](#project-structure)
+  - [Releases](#releases)
 
 ## Issue Reporting Guidelines
 
-- Always use [GitHub Issues][github-issues] to create new issues.
+- Use the [#acpaas-ui channel](https://dgpls.slack.com/messages/C4M60PQJF) on slack to ask questions about using ACPaaS UI.
```
Comment (jsprds) [inline CONTRIBUTING.md]: Now I think again about using slack as communication channel. The digipolis Slack isn't accessible for users with no digipolis.be mailbox.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -2,78 +2,97 @@
...
 
-- Checkout a topic branch from the relevant branch, e.g. `dev`, and merge back against that branch.
+- Commit your changes to a topic branch with a name matching these examples:
+  - `feature/more-cowbell`
+  - `fix/broken-cowbell`
+  - `docs/document-cowbell`
+  - For a list of suggested keywords, check the [commit message convention](./guidelines/COMMITS.md).
 
-- Work in the `src` folder and **DO NOT** check in `dist` in the commits.
+- Work in the `packages` folder and **DO NOT** check in `dist` or `examples` in the commits.
```
Comment (jsprds) [inline CONTRIBUTING.md]: It should be a mandatory step to check in examples. The example folders are used on: https://digipolisantwerp.github.io/acpaas-ui_angular/modules/avatar

Every change or new feature should be added as an example

Comment (jsebrech) [reply CONTRIBUTING.md]: I'll improve the wording. I meant the top-level examples folder which is a build artifact from the styleguide app. Maybe that folder should be placed below dist or renamed to dist-examples.

File: CONTRIBUTING.md (hunk 1 above)
Comment (jsebrech) [reply CONTRIBUTING.md]: Hmm, I forgot about that one. TBD what strategy to use. Options: open up dgpls slack, make a separate dgpls-open slack space, make a separate acpaas-ui slack space, or make a gitter space.

File: guidelines/VERSIONING.md [hunk 3]
```diff
@@ -16,17 +21,37 @@ The maintainers will edit the `CHANGELOG.md` according to the guidelines on [Kee
...
+- **Major**
 
-  When a method is removed, the major version number MUST be incremented, and there MUST be a patch or minor release of the previous major version that includes a deprecation warning.
+  When a method is removed, or its signature changes in breaking ways, the major version number MUST be incremented, and there MUST be a patch or minor release of the previous major version that includes a deprecation warning. Consensus among the maintainers should be reached before doing a new major version since it creates a maintenance burden.
 
 ## Maintaining multiple versions
 
-When multiple versions are supported, we duplicate the `master`-branch to maintain support and leave the ability to patch and publish the older supported version.
-e.g. Our library is at v4 (`master`), but v3 (`master-3.x`) is still supported.
+Previous major versions are supported with fixes for 12 months after the release of the next major version. Fixes are not automatically backported, only where necessary. Features are only added on the latest major version.
```
Comment (TriangleJuice) [inline guidelines/VERSIONING.md]: What to do when someone wants to introduce a new feature **in** an older version? E.g. when the component is part of a large application that can't be upgraded easily. Should there be specific guidelines for this case?

Comment (jsebrech) [reply guidelines/VERSIONING.md]: We can consider a new minor release on an older branch, but obviously no breaking changes are possible. Regardless the responsibility of backporting the feature lies with the app developer who needs it. I will add a note accordingly.

File: CONTRIBUTING.md (hunk 1 above)
Comment (jsebrech) [reply CONTRIBUTING.md]: Was discussed internally with the enterprise architects. The consensus was to create a separate slack space for all things open at digipolis (open source, open data, community, ...). The acpaas-ui channels will move to this new slack space.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -2,78 +2,97 @@
...
 ```
 
-There are some other scripts available in the `scripts` section of the `package.json` file.
+## Project Structure
 
-The default test script will do the following: lint with ESLint -> unit tests with coverage -> e2e tests. **Please make sure to have this pass successfully before submitting a PR.** Although the same tests will be run against your PR on the CI server, it is better to have it working locally beforehand.
+- `dist`: contains the build output for the components
+- `examples`: contains the build output for the examples
+- `packages`: the component sources
+  - `some-package/`: sources for a component package (one or more components)
```
[the remark this answers is not in the document]
Comment (jsebrech) [reply CONTRIBUTING.md]: Heb some-package en some-other-package gedaan, want twee keer your-package las nogal raar.

File: CONTRIBUTING.md (hunk 1 above)
Comment (jsebrech) [reply CONTRIBUTING.md]: Will make it a separate issue to update the slack url's to an openly accessible slack space

**Pull request merged**